source/_integrations/ipp.md

```diff
@@ -17,3 +17,5 @@ The `Internet Printing Protocol (IPP)` integration allows you to read current da
 It provides information about the printer's state and remaining ink levels.
 
 {% include integrations/config_flow.md %}
+
+Currently, this integration shows ink levels as unavailable when a printer is offline. To save the levels an automation can be used to save them into an input_number helper when the printer comes online or the level is updated. Then a template sensor can be used to display them as sensors even when the printer is offline.
```

Comment:
- Some rewording
- Template sensor sounds unneeded

```suggestion
Ink levels are shown as unavailable when a printer is offline. To save the levels for displaying during an unavailability, automation can be used to save them into an [Input Number](/integrations/input_number) helper.
```

Could we add example automation for clarity?

Reply:
Sounds good for the automation, I will add an example.

Unfortunately, the template sensor is necessary since otherwise the information is not displayed as a sensor in a lovelace card which does not make sense given it is a pure input. The only options "Slider" and "Input field" in Display mode are wrong for this kind of input_number.

Reply:
Aah, because you want is displayed as a read-only sensor. Why not use a trigger-based template sensor in that case? In such a way it gets updated each time the ink levels change? 

That would remove the need for a helper altogether.

Reply:
The honest answer is "because this is the first way I got it working". 

If I remember correctly, the issue was that if you use a template sensor, and just use a reference like '{{ states(''sensor.canon_tr8500_series_cyan'') }}', the template sensor will also show as "Unknown" and I was not familiar how to code an exception into the template sensor to not update when the update is "Unknown". The input_number gets around this because it cannot handle "Unknown".

Reply:
I can try again with the trigger-based template sensor given that my HA YAML knowledge has increased. Btw. this workaround basically solves https://github.com/home-assistant/core/issues/64545.

Reply:
Your solution also has the problem that it does not survive a restart of Home Assistant.